modules/iam-role-for-service-accounts-eks/policies.tf

```
@@ -422,6 +422,50 @@ resource "aws_iam_role_policy_attachment" "efs_csi" {
   policy_arn = aws_iam_policy.efs_csi[0].arn
 }
 
+################################################################################
+# S3 CSI Driver Policy
+################################################################################
+
+#https://github.com/awslabs/mountpoint-s3/blob/main/doc/CONFIGURATION.md#iam-permissions
+data "aws_iam_policy_document" "s3_csi" {
+  count = var.create_role && var.attach_s3_csi_policy ? 1 : 0
+
+  statement {
+    sid       = "MountpointFullBucketAccess"
+    actions   = ["s3:ListBucket"]
+    resources = var.s3_csi_bucket_arns
+  }
+
+  statement {
+    sid = "MountpointFullObjectAccess"
+    actions = [
+      "s3:GetObject",
+      "s3:PutObject",
+      "s3:AbortMultipartUpload",
+      "s3:DeleteObject"
+    ]
+    resources = formatlist("%s/*", var.s3_csi_bucket_arns)
```

this could surprise users, lets leave it to them to provide the proper arn with path prefix - we can use a separate variable for this to distinguish between the buckets for the `list` permissions and the paths for the `get`/`put`/`delete` permissions

```suggestion
    resources = var.s3_csi_path_arns
```

---

updated but I matched the pattern for coalescelist(var.s3_csi_path_arns, ["arn:${local.partition}:s3:::*/*"]) from the bucket arns. Let me know if that's an issue